core/chaincode/lifecycle/metadata_provider.go

```diff
@@ -119,14 +119,36 @@ func (mp *MetadataProvider) Metadata(channel string, ccName string, includeColle
 		spe = cauthdsl.MarshaledRejectAllPolicy
 	}
 
+	// process any existing collection endorsement policies
+	collectionPolicies := map[string][]byte{}
+	if ccInfo.Definition.Collections != nil {
```

Comment:
[Return early, return often](http://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html).

Can we please simply return the object and skip populating the collection policies if the **Definition.Collections** is nil?

Can we please switch the: if **staticCollConfig.EndorsementPolicy != nil**  to be if if **staticCollConfig.EndorsementPolicy == nil**  and then continue the loop?

Reply:
Ack